azure/services/groups/groups.go

```diff
@@ -125,8 +85,7 @@ func (s *Service) ShouldDeleteIndividualResources(ctx context.Context) bool {
 	// For ASO, "managed" only tells us that we're allowed to delete the ASO
 	// resource. We also need to check that deleting the ASO resource will really
 	// delete the underlying resource group by checking the ASO reconcile-policy.
-	spec := s.Scope.GroupSpec()
-	group := spec.ResourceRef()
+	group := s.Specs[0].ResourceRef()
```

Comment:
Even though we know there is a single group in the specs for groups, it might be best to check the length of s.Specs so we don't panic if the specs field is somehow empty

Reply:
Good callout. Instead though I think I'll refactor this to handle multiple specs a bit more natively so we don't need to index directly into this slice. This is something I was planning to do soon anyway to handle managing a vnet's resource group separately (if it's different).

Reply:
nice